File: get.sfc_anal.pl
#!/usr/bin/perl

# define system env variables
$ENV{'PATH'} = "/usr/local/bin:/usr/ucb:".$ENV{'PATH'};
$ENV{'PYTHONPATH'} = "/home/disk/bob/impacts/bin:/home/disk/shear2/brodzik/python:/usr/local/lib/python2.7/dist-packages:".$
ENV{'PYTHONPATH'};
$ENV{'LD_LIBRARY_PATH'} = "/opt/intel/compilers_and_libraries_2019.1.144/linux/compiler/lib/intel64_lin";

#$outdir = "/home/disk/bob/impacts/bin/atlantic_test";
$outdir = "/home/disk/funnel/impacts/archive/ops/sfc_anal";
$url1 = "https://ocean.weather.gov/UA/East_coast.gif";
$url2 = "https://ocean.weather.gov/UA/USA.gif";

# get date and time info
$date = `date -u "+%Y%m%d"`;
chop($date);
$hour = `date -u "+%H"`;
chop($hour);

# create outdir subdir if required
unless (-e $outdir.'/'.$date) {
    mkdir $outdir.'/'.$date;
}

# determine fileDateTime
if ($hour == '04') {
    $fileDateTime = $date.'0000';
}
elsif ($hour == '10') {
    $fileDateTime = $date.'0600';
}
elsif ($hour == '16') {
    $fileDateTime = $date.'1200';
}
elsif ($hour == '22') {
    $fileDateTime = $date.'1800';
}
else {
    print 'hour = $hour: Not one of 4 hours of interest';
    exit;
}

# download files and rename them
chdir($outdir.'/'.$date);
$cmd = 'wget '.$url1;
system($cmd);
$cmd = 'mv East_coast.gif '.$outdir.'/'.$date.'/ops.sfc_anal.'.$fileDateTime.'.atlantic.gif';
system($cmd);
$cmd = 'wget '.$url2;
system($cmd);
$cmd = 'mv USA.gif '.$outdir.'/'.$date.'/ops.sfc_anal.'.$fileDateTime.'.n_amer.gif';
system($cmd);

File: get_sfc_anal.py
#!/usr/bin/python3

import os
import sys
from datetime import datetime
from ftplib import FTP
import shutil

debug = 1
test = False
tempDir = '/tmp'
#outdir = '/home/disk/funnel/impacts/archive/ops/sfc_anal'
url1 = 'https://ocean.weather.gov/UA/East_coast.gif'
url2 = 'https://ocean.weather.gov/UA/USA.gif'
catalogPrefix = 'surface.NWS_Surface_Analysis'

# Field Catalog inputs
if test:
    ftpCatalogServer = 'ftp.atmos.washington.edu'
    ftpCatalogUser = 'anonymous'
    ftpCatalogPassword = 'brodzik@uw.edu'
    catalogDestDir = 'brodzik/incoming/impacts'
else:
    ftpCatalogServer = 'catalog.eol.ucar.edu'
    ftpCatalogUser = 'anonymous'
    catalogDestDir = '/pub/incoming/catalog/impacts'

# Open ftp connection
if test:
    catalogFTP = FTP(ftpCatalogServer,ftpCatalogUser,ftpCatalogPassword)
    catalogFTP.cwd(catalogDestDir)
else:
    catalogFTP = FTP(ftpCatalogServer,ftpCatalogUser)
    catalogFTP.cwd(catalogDestDir)

# get date and time info
dateTime = datetime.utcnow()
dateStr = dateTime.strftime("%Y%m%d")
hourStr = dateTime.strftime("%H")

# determine fileDateTime
if hourStr == '04':
    fileDateTime = dateStr+'0000'
elif hourStr == '10':
    fileDateTime = dateStr+'0600'
elif hourStr == '16':
    fileDateTime = dateStr+'1200'
elif hourStr == '22':
    fileDateTime = dateStr+'1800'
else:
    print('hour =',hourStr,': Not one of 4 hours of interest')
    sys.exit(0)

# download files, rename them and ftp them
os.chdir(tempDir)

command = 'wget '+url1
os.system(command)
catalogName = catalogPrefix+'.'+fileDateTime+'.Atlantic.gif'
shutil.move('East_coast.gif',
            catalogName)
ftpFile = open(os.path.join(tempDir,catalogName),'rb')
catalogFTP.storbinary('STOR '+catalogName,ftpFile)
ftpFile.close()
os.remove(os.path.join(tempDir,catalogName))
if debug:
    print('ftpd ',catalogName,' to NCAR FC')

command = 'wget '+url2
os.system(command)
catalogName = catalogPrefix+'.'+fileDateTime+'.North_America.gif'
shutil.move('USA.gif',
            catalogName)
ftpFile = open(os.path.join(tempDir,catalogName),'rb')
catalogFTP.storbinary('STOR '+catalogName,ftpFile)
ftpFile.close()
os.remove(os.path.join(tempDir,catalogName))
if debug:
    print('ftpd ',catalogName,' to NCAR FC')

# Close ftp connection
catalogFTP.quit()



